compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BUnionType.java

```diff
@@ -108,7 +108,9 @@ public void setNullable(boolean nullable) {
      * @return The created union type.
      */
     public static BUnionType create(BTypeSymbol tsymbol, LinkedHashSet<BType> types) {
-        LinkedHashSet<BType> memberTypes = toFlatTypeSet(types);
+        LinkedHashSet<BType> memberTypes = toFlatTypeSet(types).stream()
+                .filter(memBType -> memBType.tag != TypeTags.NEVER)
+                .collect(Collectors.toCollection(LinkedHashSet::new));
```

Comment:
Not exactly related to this PR. But can you convert the stream API usages in this class to loops as well? Should be straight forward with IntelliJ

Reply:
Okay